historian-server/src/main/java/com/hurence/webapiservice/timeseries/extractor/AbstractTimeSeriesExtracter.java

```diff
@@ -98,6 +102,13 @@ public JsonObject getTimeSeries() {
         return toReturn;
     }
 
+    private JsonArray returnPoint(Point point) {
+        if (point.hasQuality())
+            return new JsonArray().add(point.getValue()).add(point.getTimestamp()).add(point.getQuality());
+        else
+            return new JsonArray().add(point.getValue()).add(point.getTimestamp());
+    }
```

Comment:
Not sure if you handle this case elsewhere but here you should set a DEFAULT if asked. We said that by default we had quality 1 for point without quality.